packages/components/src/components/telekom/telekom-nav-item/telekom-nav-item.tsx

```diff
@@ -48,21 +50,34 @@ export class TelekomNavItem {
     if (this.variant === 'lang-switcher' && this.active) {
       toggleAriaCurrent(this.linkElement, newValue, 'true');
     }
+    if (this.variant === 'main-nav' && this.active) {
+      toggleAriaCurrent(this.linkElement, newValue, 'true');
+    }
   }
 
   connectedCallback() {
     this.activeChanged(this.active);
   }
 
+  componentDidLoad() {
+    const child = Array.from(this.hostElement.children).find((el) =>
+      el.matches('a, button')
+    );
+    const parentRole = this.hostElement.parentElement.getAttribute('role');
+    if (parentRole === 'menu') {
+      child.setAttribute('role', 'menuitem');
+    }
+  }
+
   get linkElement(): HTMLAnchorElement | null {
-    return this.hostElement.querySelector('a');
+    return this.hostElement.querySelector('a, button');
   }
 
   render() {
     return (
       // A class is used to avoid coupling styles to the tagname
       // (which can be different based on who defines it)
-      <Host class="scale-telekom-nav-item">
+      <Host class="scale-telekom-nav-item" title={this.navItemTitle}>
```

Comment:
I would prefer we remove this `navItemTitle` prop and use `title` directly in the markup. Setting `<scale-telekom-nav-item nav-item-title="foo">` is effectively the same as doing `<scale-telekom-nav-item title="Foo">`.